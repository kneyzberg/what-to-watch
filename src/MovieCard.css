.MovieCard-container {
  width: 200px;
  height: 330px;
  /* border: 1px solid black; */
}

.Movie-card-text {
  font-family: 'Scheherazade New', serif;
  font-weight: bold;
  font-size: 1.2rem;
}