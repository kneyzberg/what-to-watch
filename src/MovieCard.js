import React, { useState } from "react";
import "./MovieCard.css";

function MovieCard({ imgUrl, posterSize, movie }) {
  const [hover, setHover] = useState(false);
  const posterURL = `${imgUrl}/${posterSize}/${movie.poster_path}`;

  function mouseOn(e) {
    setHover(true);
  }

  function mouseOut(e) {
    setHover(false);
  }

  return (
    <div className="MovieCard-container">
      <img
        src={posterURL}
        alt={`Poster for ${movie.title}`}
        onMouseOver={mouseOn}
        onMouseOut={mouseOut}
        className={hover ? "Image-transparent" : ""}
      />
      <div className="Movie-card-text">{movie.title}</div>
      <div className="Movie-card-text">({movie.release_date.slice(0, 4)})</div>
    </div>
  );
}

export default MovieCard;
