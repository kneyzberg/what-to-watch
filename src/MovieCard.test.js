import MovieCard from "./MovieCard";
import React from "react";
import { render } from "@testing-library/react";

const movie = {
  title: "Testing Movie Card",
  release_date: "2021-09-22",
  poster_path: "/test.jpg"
}
const imgUrl = "testingmovies.com";
const posterSize = "w-3";

describe("Movie card test", () => {
  test("Movie Card renders correctly", () => {
    render(<MovieCard movie={movie} imgUrl={imgUrl} posterSize={posterSize}/>);
  });

  test("matches snapshot", () => {
    const { container, getByAltText } = render(<MovieCard movie={movie} imgUrl={imgUrl} posterSize={posterSize} />);
    expect(container).toMatchSnapshot();
    expect(getByAltText("Poster for Testing Movie Card")).toBeInTheDocument();
  })
});