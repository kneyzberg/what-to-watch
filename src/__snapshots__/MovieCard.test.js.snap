// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie card test matches snapshot 1`] = `
<div>
  <div>
    <div>
      Testing Movie Card
       
      2021-09-22
    </div>
    <img
      alt="Poster for Testing Movie Card"
      src="testingmovies.com/w-3//test.jpg"
    />
  </div>
</div>
`;
